www/src/components/shell/onboarding/cloud/CloudBuild.js

```diff
@@ -47,7 +47,7 @@ function CloudBuild() {
         bucketPrefix: `plural-${Math.random().toString().substring(2, 8)}`,
       }))
     }
-  }, [results])
+  }, [results, setCredentials, setDemoId, setProvider, setWorkspace])
```

Comment:
I believe the reason those dependencies weren't added was to avoid update spam issues which can happen with useEffect hooks